### ISSUE TITLE
Full Name not stored in Database when Using Apple Sign-In in React Native

### ISSUE THREAD

**Issue by VictorienDruon** [supabase/supabase-js#899]
## Description

When using Apple's native sign-in with React Native and the signInWithIdToken method, the raw_user_meta_data in the auth.users table doesn't include the newly created user's full name. Even though AppleAuthentication.signInAsync provides both the email and full name, only the email is stored in the database.

Code:
```
const credential = await AppleAuthentication.signInAsync({
  requestedScopes: [
	AppleAuthentication.AppleAuthenticationScope.FULL_NAME,
	AppleAuthentication.AppleAuthenticationScope.EMAIL,
  ],
});

console.log(credential);

if (credential.identityToken) {
  const { error } = await supabase.auth.signInWithIdToken({
	  provider: "apple",
	  token: credential.identityToken,
  });
}
```

Console log:
```
{
  authorizationCode: "**********",
  email: "**********",
  fullName: {
  familyName: "Druon",
  givenName: "Victorien",
  middleName: null,
  namePrefix: null,
  nameSuffix: null,
  nickname: null,
  },
  identityToken: "**********",
  realUserStatus: 2,
  state: null,
  user: "**********",
}
```

Supabase auth.users.raw_user_meta_data:
```
{
  "iss": "https://appleid.apple.com",
  "sub": "**********",
  "email": "**********",
  "provider_id": "**********",
  "custom_claims": {
    "auth_time": 1698918001
  },
  "email_verified": true
}
```

## To Reproduce

Follow the Supabase login with Apple with Expo React Native documentation: https://supabase.com/docs/guides/auth/social-login/auth-apple?platform=react-native

## Expected behavior

The user's full name should be added to raw_user_meta_data.

## System information
```
"dependencies": {
  ...
  "@supabase/supabase-js": "^2.31.0",
  "expo": "~49.0.5",
  "expo-apple-authentication": "~6.1.0",
  "expo-auth-session": "~5.0.2",
  "expo-router": "2.0.0",
  "react": "18.2.0",
  "react-native": "0.72.6",
  "supabase": "^1.82.2",
  ...
},
```

## Additional context
- Apple only returns the requested information on the very first authentication. The log above is what you get for the first auth.
- There is a similar issue in supabase-flutter: https://github.com/supabase/supabase-flutter/issues/545

**Comment by aalmho** [comment]
I have the same problem with auth.users not including the name of the user upon sign up. However by maintaining users in a public tabel like proposed by supabase: https://supabase.com/docs/guides/auth/managing-user-data, you can do a work around to upsert or update the profiles table when signup is successful like: 

`await supabase.from("profiles").upsert({id: user?.id, first_name: credential.fullName?.givenName, last_name: credential.fullName?.familyName}).eq("id", user?.id);`

**Comment by VictorienDruon** [comment]
Yeah, that would work, but it would be better if we didn't have to do this workaround. Because of that, we can't use supabase.auth.getUser() and have up-to-date user information. I wish signInWithIdToken would handle this automatically or at least allow us to pass additional user metadata, similar to how it works with supabase.auth.signUp()

**Comment by Schotsl** [comment]
I'm running into the same issue. While using the metadata from the JavaScript call does work. It adds a unneeded step to the signup process.

**Comment by ryonwhyte** [comment]
Ran into the same issue here. Apple is the last I am integrating of a few social networks. Now, I'll need to write some sort of workaround into my login process.

**Comment by dylandrendel** [comment]
Same issue for me in Flutter. When using apple OAuth sign in for android, the name is stored correctly, but for native it is not. Will try the workaround listed above.

**Comment by vbylen** [comment]
can confirm this is still an issue

**Comment by guilhermefc** [comment]
+1

**Comment by stephanrueschenbaum** [comment]
+1

**Comment by andyngdz** [comment]
I'm also getting this issue

**Comment by TuanTran-Tsk** [comment]
any update ??

**Comment by kangmingtay** [comment]
hey everyone, as a workaround, you can call `supabase.auth.updateUser` to update the `raw_user_meta_data` to include the full_name of the user

**Comment by FezVrasta** [comment]
I'm unable to retrieve it even when I look into the credential object. It's always `null` even though I removed the Expo app from the "Sign in with Apple" list.

**Comment by selimb86** [comment]
Same with Flutter

**Comment by azizsafudin** [comment]
> I'm unable to retrieve it even when I look into the credential object. It's always `null` even though I removed the Expo app from the "Sign in with Apple" list.

It's impossible to get it again. You'd need to delete your Apple App ID and recreate an App ID, service ID, certificates, etc.

**Comment by AoMafura** [comment]
Same here

**Comment by soybeins** [comment]
> hey everyone, as a workaround, you can call `supabase.auth.updateUser` to update the `raw_user_meta_data` to include the full_name of the user

Seems that this one currently is the most viable solution to bypass this issue.

**Comment by KhaleelSH** [comment]
In my flutter code I fixed it by adding the following code after `signInWithIdToken`:
```
    await supabase.auth.updateUser(
      UserAttributes(
        data: {
          'given_name': credential.givenName,
          'family_name': credential.familyName,
        },
      ),
    );
```

**Comment by wycks** [comment]
This seems to be still broken? Using the flutter docs here https://supabase.com/docs/guides/auth/social-login/auth-apple?queryGroups=platform&platform=flutter  it mentions `AppleIDAuthorizationScopes.fullName,`  I just created a brand new apple ID + email + first and last name,  but the name is NOT not stored in the Auth DB, nor on any return strings when the code fires, the email returns, but the name is empty.

**Comment by TizianoCoroneo** [comment]
It is still broken. I'm using the native iOS Supabase SDK, requesting the `.fullName` scope, and the auth DB doesn't get the display name, even when fully resetting by revoking permissions for SIWA on device AND deleting the user in the DB.

**Comment by nyeu** [comment]
This is also happening for native iOS

**Comment by grdsdev** [comment]
Hi everyone, Guilherme from the Supabase SDK Team here!

I wanted to share some important info about this, as there might be a few different things floating around.

I’m not sure if Apple’s identity token used to include the full name in its claims, but it doesn’t anymore. So, it’s just not possible for the Supabase server to automatically fill in the user’s name field when we only have the identity token.

The best way to handle this is to use the `updateUser` method to insert the name Apple returns into the user’s table. Just a heads-up: Apple only gives you the full name the first time you sign up (during sign-up). After that, any Apple sign-ins will return null in the full name field.

You can check out the [Swift example](https://github.com/supabase/supabase-swift/blob/main/Examples/Examples/Auth/SignInWithApple.swift#L70-L82) for the recommended way to populate the full name on Apple sign-in. Even though it’s in Swift, it’s super easy to adapt to Expo/Flutter or any other mobile platform.

I’ll also chat with the docs team to make sure this gets more attention. Plus, I’ll talk to the Auth team about any other potential solutions to this limitation.

Thanks!

**Comment by nyeu** [comment]
Hi @grdsdev, 
Thanks for the explanation! The issue is that the resopnse from the `SignInWithApple` button doesn't return the full name, not even the first time you sign in. The fullName is requested and it even appears in the Apple sign in sheet but it's never returned by the SDK. I have a response example if you are interested.

**Comment by azizsafudin** [comment]
Thanks @grdsdev, I think updating the docs would be sufficient for now!

**Comment by mandarini** [comment]
Hi @VictorienDruon, thank you for opening this issue! Quick follow-up on Guilherme's note from October. After tracing through the SDK and auth server, this is expected behaviour, and the docs work has since landed.

Apple ships the user's name as a separate field on the native sign-in response, not as a claim in the identity token JWT. `signInWithIdToken` only forwards the token to the auth server, and `raw_user_meta_data` is populated from the JWT claims, which do not include a name. The same is true on Flutter and native iOS.

Worth flagging: Apple only returns the full name on the very first authorization. Subsequent sign-ins return null. To reset, the user has to revoke the app under iOS Settings, Apple ID, Sign in with Apple.

The fix is to capture the name from the Apple credential and call `updateUser` right after `signInWithIdToken`:

```ts
if (!error && credential.fullName) {
  await supabase.auth.updateUser({
    data: {
      full_name: [credential.fullName.givenName, credential.fullName.familyName]
        .filter(Boolean)
        .join(' '),
      given_name: credential.fullName.givenName,
      family_name: credential.fullName.familyName,
    },
  })
}
```

The Apple sign-in guide at https://supabase.com/docs/guides/auth/social-login/auth-apple now covers this for React Native, Flutter, Swift, Kotlin, and Sign in with Apple JS (landed in supabase/supabase#39672).

Closing as addressed. If anything in the docs is unclear or the `updateUser` flow is not working for you, please open a fresh issue with a repro and I will pick it up.
